

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.vmware.pyvcloud.provider";
option java_outer_classname = "PyVcloudProviderProto";

package pyvcloudprovider;

// Interface exported by the server.
service PyVcloudProvider {
//Tenant Loging to VCD  
  rpc Login(TenantCredentials) returns ( LoginResult ) {}
  rpc isPresentCatalog (  Catalog ) returns ( IsPresentCatalogResult ) {}
  rpc CreateCatalog ( Catalog ) returns ( CreateCatalogResult )  {}
  rpc DeleteCatalog ( Catalog ) returns ( DeleteCatalogResult )  {}

}

//Tenant VCD crendentials
message TenantCredentials {
  string username = 1;
  string password = 2;
  string org = 3;
  string ip = 4;

}


message LoginResult{
	string token = 1;
}

message Catalog {
	string name = 1;
	string description = 2;
	bool shared = 3;
}

message IsPresentCatalogResult{

	bool present=1;
}

message CreateCatalogResult{
	bool created = 1;
}

message DeleteCatalogResult{
        bool deleted = 1;
}
